cmd/panfs.go

```diff
@@ -1224,7 +1308,7 @@ func (fs *PANFSObjects) DeleteObject(ctx context.Context, bucket, object string,
 
 	if bucket != minioMetaBucket {
 		// Delete the metadata object.
-		err = fsDeleteFile(ctx, minioMetaBucketDir, fsMetaPath)
+		err = fsDeleteFile(ctx, pathJoin(bucketPanfsDir, panfsMetaDir, objMetadataDir), fsMetaPath)
```

Comment:
You have changed the logic from the original here. Why? If this is intended, maybe it makes sense to add a comment?
If I am not mistaken, analogously to the original version we would have:
```go
		err = fsDeleteFile(ctx, bucketPanfsDir, fsMetaPath)
```
?

Reply:
In this line we are deleting the object metadata which is not in the default location (as this change provides a new location for object metadata - but the real data are still storing on the standard location)

Reply:
Yes, that's right. The `fsDeleteFile()` function removes the parent directories up to the directory provided as the second argument. After your change there are going to be left more empty directories in the place where the metadata is stored than before the change. Is my reasoning right?

Reply:
I'm not sure. This function should delete directories with object metadata including object metadata directory (e.g. <bucket>/.s3/metadata/**\<object\>/panfs.json** the path in bold deleted when delete object).

There is a test run:
```
dmitri_zavadski@EPBYBREW0207:~/projects/minio-project/minio-mc$ ./mc mb local/inner
Bucket created successfully `local/inner`.
dmitri_zavadski@EPBYBREW0207:~/projects/minio-project/minio-mc$ ./mc cp ~/Downloads/2212-3709_done.zip local/inner
...i/Downloads/2212-3709_done.zip: 553.11 KiB / 553.11 KiB ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 46.88 MiB/s 0sdmitri_zavadski@EPBYBREW0207:~/projects/minio-project/minio-mc$ ./mc ls local/inner
[2023-01-18 18:18:08 +03] 553KiB STANDARD 2212-3709_done.zip
dmitri_zavadski@EPBYBREW0207:~/projects/minio-project/minio-mc$ ll ~/minio_panfs_gateway/
total 16
drwxrwxr-x  4 dmitri_zavadski dmitri_zavadski 4096 сту 18 18:18 ./
drwxr-xr-x 72 dmitri_zavadski dmitri_zavadski 4096 сту 18 17:28 ../
drwxrwxr-x  3 dmitri_zavadski dmitri_zavadski 4096 сту 18 18:18 inner/
drwxrwxr-x  6 dmitri_zavadski dmitri_zavadski 4096 сту 18 18:18 .minio.sys/
dmitri_zavadski@EPBYBREW0207:~/projects/minio-project/minio-mc$ ll ~/minio_panfs_gateway/inner/.s3/metadata/
total 12
drwxrwxr-x 3 dmitri_zavadski dmitri_zavadski 4096 сту 18 18:18 ./
drwxrwxr-x 3 dmitri_zavadski dmitri_zavadski 4096 сту 18 18:18 ../
drwxrwxr-x 2 dmitri_zavadski dmitri_zavadski 4096 сту 18 18:18 2212-3709_done.zip/
dmitri_zavadski@EPBYBREW0207:~/projects/minio-project/minio-mc$ ./mc rm local/inner/2212-3709_done.zip
Removed `local/inner/2212-3709_done.zip`.
dmitri_zavadski@EPBYBREW0207:~/projects/minio-project/minio-mc$ ./mc ls local/inner
dmitri_zavadski@EPBYBREW0207:~/projects/minio-project/minio-mc$ ll ~/minio_panfs_gateway/inner/.s3/metadata/
total 8
drwxrwxr-x 2 dmitri_zavadski dmitri_zavadski 4096 сту 18 18:18 ./
drwxrwxr-x 3 dmitri_zavadski dmitri_zavadski 4096 сту 18 18:18 ../
```

The `metadata` directory is not deleted - the object metadata directory only should be